libs/Microsoft.MixedReality.WebRTC.Native/include/interop_api.h

```diff
@@ -70,23 +70,116 @@ MRS_API void MRS_CALL mrsCloseEnum(mrsEnumHandle* handleRef) noexcept;
 // Interop
 //
 
+struct mrsAudioTransceiverConfig;
+struct mrsVideoTransceiverConfig;
+struct mrsRemoteAudioTrackConfig;
+struct mrsRemoteVideoTrackConfig;
 struct mrsDataChannelConfig;
 struct mrsDataChannelCallbacks;
 
+// Handles to interop objects (internal implementation)
+
+/// Opaque handle to a native PeerConnection interop object.
+using mrsPeerConnectionHandle = void*;
+
+/// Opaque handle to a native MediaTrack interop object.
+using mrsMediaTrackHandle = void*;
+
+/// Opaque handle to a native AudioTransceiver interop object.
+using mrsAudioTransceiverHandle = void*;
+
+/// Opaque handle to a native VideoTransceiver interop object.
+using mrsVideoTransceiverHandle = void*;
+
+/// Opaque handle to a native LocalAudioTrack interop object.
+using mrsLocalAudioTrackHandle = void*;
+
+/// Opaque handle to a native LocalVideoTrack interop object.
+using mrsLocalVideoTrackHandle = void*;
+
+/// Opaque handle to a native RemoteAudioTrack interop object.
+using mrsRemoteAudioTrackHandle = void*;
+
+/// Opaque handle to a native RemoteVideoTrack interop object.
+using mrsRemoteVideoTrackHandle = void*;
+
+/// Opaque handle to a native DataChannel interop object.
+using mrsDataChannelHandle = void*;
+
+/// Opaque handle to a native ExternalVideoTrackSource interop object.
+using mrsExternalVideoTrackSourceHandle = void*;
+
+// Handles to wrapper objects
+
 /// Opaque handle to the interop wrapper of a peer connection.
 using mrsPeerConnectionInteropHandle = void*;
 
+/// Opaque handle to the interop wrapper of an audio transceiver.
+using mrsAudioTransceiverInteropHandle = void*;
+
+/// Opaque handle to the interop wrapper of a video transceiver.
+using mrsVideoTransceiverInteropHandle = void*;
+
+/// Opaque handle to the interop wrapper of a local audio track.
+using mrsLocalAudioTrackInteropHandle = void*;
+
 /// Opaque handle to the interop wrapper of a local video track.
 using mrsLocalVideoTrackInteropHandle = void*;
 
+/// Opaque handle to the interop wrapper of a remote audio track.
+using mrsRemoteAudioTrackInteropHandle = void*;
+
+/// Opaque handle to the interop wrapper of a remote video track.
+using mrsRemoteVideoTrackInteropHandle = void*;
+
 /// Opaque handle to the interop wrapper of a data channel.
 using mrsDataChannelInteropHandle = void*;
 
+/// Callback to create an interop wrapper for an audio transceiver.
+/// The callback must return the handle of the created interop wrapper.
+using mrsAudioTransceiverCreateObjectCallback =
```

Comment:
It might be worth explaining _when_ each of these "CreateObject" callbacks is called (which I assume is when an object is created following an action by the remote peer, but I am not so sure because there are many!)

Reply:
Is it though? You are supposed to provide a callback to create a wrapper, that should be generic enough, without really needing to understand why/when it's needed, no? The implementation will call it when it needs to create a wrapper to associate with a new implementation object, there is not much more to know I think.

Reply:
Fair enough. The specific thing I find confusing is that some add/create functions expect the upper layer to pass an interop handle there, and it's not specified if these callbacks will be called in that case or not (if they are, it sounds redundant; if they aren't, we probably should make the distinction in the comments.)

Reply:
This should change as part of #207, resolving for now.